src/base/TcpSocketHandler.cpp

```diff
@@ -12,7 +12,7 @@ int TcpSocketHandler::connect(const SocketEndpoint &endpoint) {
   memset(&hints, 0, sizeof(addrinfo));
   hints.ai_family = AF_UNSPEC;
   hints.ai_socktype = SOCK_STREAM;
-#if __NetBSD__
+#if defined(__NetBSD__) || defined(__ANDROID__)
   hints.ai_flags = (AI_CANONNAME | AI_ADDRCONFIG);
```

Comment:
[As noted on SO, Android doesn't support these flags](https://stackoverflow.com/questions/39674147/getaddrinfo-on-android-returning-error-eai-badflags) just like BSD, which is where its source comes from.